packages/edit-site/src/components/post-fields/index.js

```diff
@@ -235,7 +235,7 @@ function usePostFields( viewType ) {
 			{
 				label: __( 'Author' ),
 				id: 'author',
-				getValue: ( { item } ) => item._embedded?.author[ 0 ]?.name,
```

Comment:
As far as I'm aware, we didn't really use this function. The implementation is also wrong, because the `getValue` should return `item.author` (an integer, what the dataset uses) instead of a name (the display name is a `render` concern).

Reply:
Don't we have a filter that uses this?

Reply:
It's not necessary for filters, using the integer would be enough. However! Your comment makes me realize this is actually necessary for _sorting_: we should sort authors by name not by the id assigned by WordPress.

I'm going to bring back this, but sorting is not working on `trunk`. We need to look at that separately. There's a few issues (author & dates not sorting properly, the sort config in the table headers is not cleared properly):

https://github.com/user-attachments/assets/fa98a60b-0b48-47e9-8350-0241d66e6892

Reply:
This actually raises an interesting question: if we have types but sorting depends on getValue, we can end up with the following:

- type is `integer`
- but `getValue` returns a string

How do we make the [sort utility](https://github.com/WordPress/gutenberg/blob/add/author-to-quick-edit/packages/dataviews/src/filter-and-sort-data-view.ts#L136) work based on types in this scenario?

It seems we shouldn't use `getValue` for sorting and actually provide a custom sort utility if the field needs it. In that case, the filterSort utility would use the custom sort if it exists, otherwise would default to the sort defined by the field type. Thoughts?

_(nothing to do in this PR, as I brought back the `getValue` function https://github.com/WordPress/gutenberg/pull/63983/commits/216234ecd45d8f8e297f7efd44106578d46bf241)_

Reply:
I've looked a bit more into this and it turns out `getValue` is not used for sorting either: we delegate on [the REST API](https://github.com/WordPress/gutenberg/blob/add/author-to-quick-edit/packages/edit-site/src/components/post-list/index.js#L164) for filter/sort/rest in this page.

The issues I've seen with sorting in `trunk` are:

- author is sorted by creation id, not by name
- date: I'm not sure why it doesn't work but it's unrelated.

I removed the `getValue` to proceed with this PR.